### PR TITLE
Kerberos Hive integration

### PR DESCRIPTION
**Pull request by petroav**
This PR is just for code discussions.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java [hunk 1]
```diff
@@ -376,8 +377,13 @@ public static void renameDirectory(HdfsEnvironment hdfsEnvironment, String schem
     public static Path createTemporaryPath(HdfsEnvironment hdfsEnvironment, Path targetPath)
     {
         // use a per-user temporary directory to avoid permission problems
-        // TODO: this should use Hadoop UserGroupInformation
-        String temporaryPrefix = "/tmp/presto-" + StandardSystemProperty.USER_NAME.value();
+        String temporaryPrefix;
+        try {
+            temporaryPrefix = "/tmp/presto-" + UserGroupInformation.getCurrentUser().getUserName();
```
Comment (billonahill) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java:382]: Wouldn't this tmp dir need to be in the users home dir, or somewhere where they have write permissions?

File: presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java [hunk 2]
```diff
@@ -139,7 +142,18 @@ public void start(HiveSplitSource splitSource)
...
-            ResumableTasks.submit(executor, new HiveSplitLoaderTask());
+            UserGroupInformation currentHdfsUser = getCurrentHdfsUser();
+            ResumableTasks.submit(executor, new HiveSplitLoaderTask(currentHdfsUser));
+        }
+    }
+
+    private UserGroupInformation getCurrentHdfsUser()
+    {
+        try {
+            return UserGroupInformation.getCurrentUser();
```
Comment (billonahill) [inline presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java:153]: Wouldn't this return the daemon user and not the querying user? I'd think you'd need to call HadoopAuthentication.getUserGroupInformation(queryingUser). Or have you already changed the current user at a higher level?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java (hunk 1 above)
Comment (billonahill) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java:382]: Answering my own question here since /tmp is executable by all. The issue we run into is that /tmp and /user are on different viewfs filesystems so we can't easily copy from /tmp to /user. Changing this to create the tmp/ dir in the user home could be a different patch of ours.

Comment (losipiuk) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java:382]: @billonahill. I wanted to change the code  it as you suggest. 
But as I read the codebase it seems that temporary file is not created in `/tmp` 
but in `$targetPath/tmp/`. This is by definition on the same filesystem as `$targetPath`. I will debug yet - to verify - but it seems like you should not have the problem you are describing. Am I missing somethign?

File: presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java (hunk 2 above)
Comment (losipiuk) [inline presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java:153]: Yeah - this section is called from top-level thread initiating split-loading process. And it is wrapped in `doAs` section by the caller.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java (hunk 1 above)
Comment (losipiuk) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java:382]: Never mind. Preparing patch right now :)

Comment (losipiuk) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java:382]: @billonahill you can give this a try: https://github.com/Teradata/presto/pull/109

Also if you could explain in some more datail the problem you were facing - so I could put that in commit message - it would be helpful.
The old code resolved `/tmp` in context of `targetPath`. So there should not be file system mismatch. The problem which could appear was that there was no `/tmp` on the target file system. Is that what you saw?
